Prevent exhaustion of DB connections

bdc34 opened the pull request:
RDS has 60 connection limit but 8 processes x 100 ugreen “cores” (see uwsgi docs) => 800 ugreen “cores”!

Reducing number of green cores to avoid running out of connections.

https://arxiv-org.atlassian.net/browse/ARXIVNG-2727

Includes other changes from Matt.

mattbierbaum committed:
making models work with mysql (date variations)

mattbierbaum committed:
creating docker compose services with redis and mysql

mattbierbaum committed:
bootstraping databases without alembic (b/c sessions)

mattbierbaum committed:
repackaging for docker / k8s

mattbierbaum committed:
removing old requirements (pipenv now)

mattbierbaum committed:
clarifying conf

mattbierbaum committed:
turning off boostrap in uwsgi

mattbierbaum committed:
updating ratelimit for clients

mattbierbaum committed:
limiting client token lifetimes

mattbierbaum committed:
setting lifetime to 4 hours

bdc34 committed:
Prevent exhaustion of DB connections

RDS has 60 connection limit but 8 processes x 100 ugreen “cores” (see uwsgi docs) => 800 ugreen “cores”!

Reducing number of green cores to avoid running out of connections.

https://arxiv-org.atlassian.net/browse/ARXIVNG-2727

erickpeirson commented on uwsgi.ini:
```diff
@@ -0,0 +1,14 @@
+http-socket = :8000
+wsgi-file = uwsgi.py
+processes = 8
+threads = 1
```
That'll definitely scale it back

erickpeirson commented:
Looks like the travis config expects to find ``dev-requirements.txt``, which is deleted in this diff

erickpeirson commented:
The change to uwsgi.ini will certainly limit the number of green threads, and therefore should limit the number of database connections if pooling is working correctly. 

How does this play out when you deploy this new configuration in the development namespace and throw load at it?

bdc34 committed:
Changing uwsgi and SQLAlchemy config to deal with mysql connections

Attempting to reduce the number of connections to a reasonable
limit while still handling traffic.

This was tested with a 4 core laptop with siege -c 40 and worked
fine. The number of connections to MySQL remained under 60.

ARXIVNG-2727

bdc34 committed:
Gets API_KEY from env vars for easier deploy

Also switches to UUID instead of incremented IDs. But it seems the IDs
are not used.

Adds retrys to writes to DB.

Falls back to just sending the ADS token even when it cannot write to
DB.

ARXIVNG-3794

bdc34 committed:
Dealing with dependency decay

alembic was pulling in a version of sqlalchemy that conflicted with
adsmutils

bdc34 committed:
Changes to how API response is handled

Getting API_TOKEN from env vars

Changes to docker file
